File: arch/src/main/java/com/github/lany192/box/activity/ModelActivity.java
package com.github.lany192.box.activity;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.github.lany192.box.mvvm.BaseViewModel;

public class ModelActivity extends BasicActivity {

    protected <T extends BaseViewModel> T getViewModel(@NonNull Class<T> modelClass) {
        return new ViewModelProvider(this).get(modelClass);
    }
}

File: arch/src/main/java/com/github/lany192/arch/activity/ModelActivity.kt
package com.github.lany192.arch.activity

import android.os.Bundle
import androidx.annotation.CallSuper
import androidx.lifecycle.ViewModel
import androidx.viewbinding.ViewBinding

/**
 * ViewBinding实现基类
 */
abstract class ModelActivity<VM : ViewModel, CVB : ViewBinding, TVB : ViewBinding> :
    BindingActivity<CVB, TVB>() {
    lateinit var viewModel: VM

    @CallSuper
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel = createDefaultViewModel(getClass(0))
    }

    open fun createDefaultViewModel(clazz: Class<VM>): VM {
        return getViewModel(clazz)
    }
}